cloudera/datadog_checks/cloudera/api_client_v7.py

```diff
@@ -0,0 +1,131 @@
+from concurrent.futures import ThreadPoolExecutor
+
+import cm_client
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.cloudera.api_client import ApiClient
+from datadog_checks.cloudera.entity_status import ENTITY_STATUS
+from datadog_checks.cloudera.metrics import NATIVE_METRICS, TIMESERIES_METRICS
+
+from .common import CLUSTER_HEALTH, HOST_HEALTH
+
+
+class ApiClientV7(ApiClient):
+    def __init__(self, check, api_client):
+        super(ApiClientV7, self).__init__(check, api_client)
+
+    def collect_data(self):
+        clusters_resource_api = cm_client.ClustersResourceApi(self._api_client)
+        read_clusters_response = clusters_resource_api.read_clusters(cluster_type='any', view='full')
+        self._log.debug("Cloudera full clusters response:\n%s", read_clusters_response)
+
+        # Use len(read_clusters_response.items) * 3 workers since
+        # for each cluster, we are executing 3 tasks in parallel.
+        with ThreadPoolExecutor(max_workers=len(read_clusters_response.items) * 3) as executor:
+            for cluster in read_clusters_response.items:
+                cluster_name = cluster.name
+
+                tags = self._collect_cluster_tags(cluster, self._check.config.tags)
+
+                executor.submit(self._collect_cluster_metrics, cluster_name, tags)
+                executor.submit(self._collect_cluster_service_check, cluster, tags)
```

Comment:
Do we need parallelization if this does no IO?

Reply:
Since `_collect_cluster_service_check()` is a fairly small function and it doesn't call the API, I think we can get rid of the executor for this.